src/Components/Recommendation/Recommendation.cy.js

```diff
@@ -109,6 +118,101 @@ describe('recommendation page for enabled recommendation with clusters enabled a
       });
     cy.ouiaId('recommendation-disable').should('exist');
   });
+
+  it('all clusters can be enabled at once', () => {
+    cy.ouiaId('hosts-acked')
+      .ouiaId('enable')
+      .click()
+      .then(() => {
+        // wait for the expected number of network calls
+        for (const cluster of clusterDetails.data.disabled) {
+          cy.wait('@enableRequest').then((xhr) => {
+            expect(xhr.request.url).to.contain(cluster.cluster_id);
+          });
+        }
+      });
+  });
+
+  describe('disabled clusters modal', () => {
+    it('can be displayed', () => {
+      cy.ouiaId('hosts-acked').ouiaId('view-clusters').click();
+      cy.ouiaId('hosts-disabled').should('exist');
+    });
+    it('has expected clusters', () => {
+      cy.ouiaId('hosts-acked').ouiaId('view-clusters').click();
+      cy.ouiaId('hosts-disabled')
+        .find(TBODY)
+        .find(ROW)
+        .should('have.length', clusterDetails.data.disabled.length);
+      const clusters = _.map(
+        clusterDetails.data.disabled,
+        (it) => it.cluster_name || it.cluster_id
+      );
+      cy.get(`td[data-label="Cluster name"]`)
+        .then(($els) => {
+          return _.map(Cypress.$.makeArray($els), 'innerText');
+        })
+        .should('deep.equal', clusters);
+    });
+  });
+});
+
+describe('recommendation page for enabled recommendation without disabled clusters', () => {
+  let data = _.cloneDeep(clusterDetails.data);
+  data.disabled = [];
+  beforeEach(() => {
+    mount(
+      <MemoryRouter>
+        <Intl>
+          <Provider store={getStore()}>
+            <Recommendation
+              rule={{ ...defaultPropsRule }}
+              ack={{ ...defaultPropsAck, data: undefined }}
+              clusters={{ ...defaultPropsClusterDetails, data: data }}
+              match={{ params: { recommendationId: 'X' } }}
+            />
+          </Provider>
+        </Intl>
+      </MemoryRouter>
+    );
+
+    cy.intercept('POST', '/api/insights-results-aggregator/v2/ack', {
```

Comment:
I though they need to be defined into a `before` or `beforeEach`, but I can try to move them

Reply:
The cannot be inside `describe` but inside a `before`. However, I believe that doing so, if you try to get all network request within a test, it will fail because it will also account for requests in other `it` tests under the same `describe`. What we can do to speed up, is to just add the `before` on the `it` that really require it, but that could mean some duplicated code. What is your preference? Have it only once in the `beforeEach` at the cost of speed, or just define it where needed at the cost of repeated code?

Reply:
Yes, this is true that more tests can make the same (maybe different in some arguments) requests. But they shouldn't fail, because there are no such tests that would refer to the same endpoint but would expect different responses. I mean we everywhere set 200 or other successful responses. Until some tests need 404 for example, I think we are safe.

Reply:
If ^ is correct, then I think `before` block is what we are looking for :)

Reply:
Do you mean the `before` block at the `describe` level? Or at each `it` level?

Reply:
At the `describe` level. The `before` block should be executed only once before the first nested `it` will be run

Reply:
Then some tests will fail. E.g. here we have a test that is

```
for (const cluster of clusterDetails.data.disabled) {
          cy.wait('@enableRequest').then((xhr) => {
            expect(xhr.request.url).to.contain(cluster.cluster_id);
          });
```

it will fail because it also account for other such calls made by other tests, so we will have to build a more complex checks. I also remember that there is also another test somewhere else that check all network calls perform and assert they are in the right number (one per cluster). That test will also be affected by that change.

In order to keep tests for isolated and with simpler checks, I think it would be better to keep in `beforeEach` as is or move to each test.

Reply:
Ah, okay I found this in docs
![image](https://user-images.githubusercontent.com/31385370/182632900-c6e44edf-76a1-4067-9f24-698933dc5c95.png)

Reply:
Then yes, the proper way is to initialize intercepts before each test (= current solution is OK). Closing it 👍🏼